webrtc/RTCPeerConnectionIceEvent-constructor.html

```diff
@@ -1,34 +1,148 @@
+  */
+  assert_equals(event.candidate, null);
+  assert_equals(event.url, null);
+
+  /*
+  Firing an RTCPeerConnectionIceEvent event named e with an RTCIceCandidate candidate means that an event with the name e, which does not bubble (except where otherwise stated) and is not cancelable (except where otherwise stated),
+  */
+  assert_false(event.bubbles);
+  assert_false(event.cancelable);
+
+}, "RTCPeerConnectionIceEvent with no eventInitDict (default)");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type", {});
+
+  /*
+  candidate of type RTCIceCandidate, readonly, nullable
+  url of type DOMString, readonly, nullable
+  */
+  assert_equals(event.candidate, null);
+  assert_equals(event.url, null);
+
+  /*
+  Firing an RTCPeerConnectionIceEvent event named e with an RTCIceCandidate candidate means that an event with the name e, which does not bubble (except where otherwise stated) and is not cancelable (except where otherwise stated),
+  */
+  assert_false(event.bubbles);
+  assert_false(event.cancelable);
+
+}, "RTCPeerConnectionIceEvent with empty object as eventInitDict (default)");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type", {
+    candidate: null
+  });
+  assert_equals(event.candidate, null);
+}, "RTCPeerConnectionIceEvent.candidate is null when constructed with { candidate: null }");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type", {
+    candidate: undefined
+  });
+  assert_equals(event.candidate, null);
+}, "RTCPeerConnectionIceEvent.candidate is null when constructed with { candidate: undefined }");
+
+
+/*
+
+4.8.1 RTCIceCandidate Interface
+
+The RTCIceCandidate() constructor takes a dictionary argument, candidateInitDict, whose content is used to initialize the new RTCIceCandidate object. When run, if both the sdpMid and sdpMLineIndex dictionary members are null, throw a TypeError.
+*/
+let candidate = "";
+let sdpMid = "sdpMid";
+let sdpMLineIndex = 1;
+let ufrag = "";
+let url = "foo.bar";
+
+test(() => {
+  const iceCandidate = new RTCIceCandidate({ candidate, sdpMid, sdpMLineIndex, ufrag });
+  const event = new RTCPeerConnectionIceEvent("type", {
+    candidate: iceCandidate,
+    url,
+  });
+
+  assert_equals(event.candidate, iceCandidate);
+  assert_false(event.bubbles);
+  assert_false(event.cancelable);
+}, "RTCPeerConnectionIceEvent with RTCIceCandidate");
+
+test(() => {
+  const plain = { candidate, sdpMid, sdpMLineIndex, ufrag };
+  assert_throws(new TypeError(), () => new RTCPeerConnectionIceEvent("type", { candidate: plain }));
+}, "RTCPeerConnectionIceEvent with non RTCIceCandidate object throws");
+
+test(() => {
+  /*
+  When firing an RTCPeerConnectionIceEvent event that contains a RTCIceCandidate object, it must include values for both sdpMid and sdpMLineIndex.
+  */
+
+  assert_throws(new TypeError(), () => {
+    new RTCPeerConnectionIceEvent("type", {
+      candidate: new RTCIceCandidate({ candidate, sdpMid, ufrag })
```

Comment:
> The spec also does not require TypeError to be thrown.

IIRC, I assumed it was implied because the other `sdpMid` and `sdpMLineIndex` construction restriction is enforced with a TypeError, but you're absolutely right—this is not explicit! Thanks for that bug as well :)